apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/mechanism_apic.py

```diff
@@ -197,21 +197,61 @@ def __init__(self):
             ofcst.AGENT_TYPE_OPFLEX_OVS)
         ha_ip_db.HAIPOwnerDbMixin.__init__(self)
 
-    def _is_dvs_vif_type(self, context, agent):
-        """Return if this port is a DVS vif
+    def _agent_bind_port(self, context, agent_list):
+        """Attempt port binding per agent.
 
-           We need to bind the port as a DVS VIF type
-           when the port belongs to nova, and when there's
-           an OpFlex agent on that (compute) host that's told
-           us it's supporting a VMware hypervisor.
+           Perform the port binding for a given agent.
+           Returns True if bound successfully.
         """
-        port = context.current
-        if port['device_owner'].startswith('compute:'):
-            hv_type = agent['configurations'].get('hypervisor_type', None)
-            if hv_type and hv_type == acst.HYPERVISOR_VCENTER:
-                return True
+        for agent in agent_list:
+            LOG.debug("Checking agent: %s", agent)
+            if agent['alive']:
+                for segment in context.segments_to_bind:
+                    if self.try_to_bind_segment_for_agent(context, segment,
+                                                          agent):
+                        LOG.debug("Bound using segment: %s", segment)
+                        return True
+            else:
+                LOG.warning(_("Refusing to bind port %(pid)s to dead agent: "
+                              "%(agent)s"),
+                            {'pid': context.current['id'], 'agent': agent})
         return False
 
+    def bind_port(self, context):
+        """Get port binding per host.
+
+           Overriding the superclass implementation
+           in order to support multiple agent types
+           in a single mechanism driver (DVS and OpFlex)
```

Comment:
Does this MD class still actually use anything from AgentMechanismDriverBase? If not, I'd remove the unneeded base class and update this comment to state that this implementation of bind_port() is similar to the one in AgentMechanismDriverBase, but supports multiple L2 agent types.